### PR TITLE
Fix: crm_mon: rhbz#1625231 - disable fence-history in file-mode

### PR DESCRIPTION
**Pull request by wenningerk**
If cib is read from file fence-history & interactive mode don't
make sense

Instead of switching to one-shot we could fix interactive mode on files in a way that we don't bail out on failure of notifications.
And it might be worth it to add a possibility to cib_file.c that allows us to query if the cib would come from a file and if so if it is still the live cib. So we wouldn't have to check for the environment everywhere cib is used and where it would make a difference if it comes from a file or if it is live.

### CODE REVIEW

**Comment by kgaillot** [comment]
I think the hidden option -x is just there for testing. I doubt it gives any advantage over CIB_file, so maybe we should deprecate it and eventually remove it. Maybe when -x is used, we should put CIB_file into the environment and let it be handled normally.

With that, we would always do cib_new(), and so maybe after that, we could filter one-shot, fence-history,etc. based on the CIB variant.

**Comment by wenningerk** [comment]
Not tested all error-cases so far but for the basic idea ...

**Comment by kgaillot** [comment]
Looks good to me

File: tools/crm_mon.c [hunk 1]
```diff
@@ -124,6 +123,7 @@ static int fence_history_level = 1;
 static gboolean print_brief = FALSE;
 static gboolean print_pending = TRUE;
 static gboolean print_clone_detail = FALSE;
+static gboolean console_initialized = FALSE;
```
[the remark this answers is not in the document]
Comment (wenningerk) [reply tools/crm_mon.c]: atm it probably even breaks compiliation without curses.
was meant to serve as a flag for any output-initialization that
needs deinitialization. but it is probaby wiser to do that
explicitly for curses and have an extra flag for other things
to come.

**Pull request merged**